### PR TITLE
Chore: Dehydrate share extension from rocketchat.js

### PR DESCRIPTION
**Pull request by gerzonc**
## Proposed changes

Dehydrate share extension (`shareExtensionInit` and `closeShareExtension` methods) from `rocketchat.js`; part of the migration to TypeScript
## How to test or reproduce

- We need to test share extension heavily
  * multiple servers (and switching between them)
  * after sharing something, go back to the app and check if the connection is active and on the correct server
  * share from `+` button as well
  * test Android and iOS carefully, as they handle share extension differently
  * different file types
- Quick general testing
  * we changed the way sdk works, so it's important

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] Improvement (non-breaking change which improves a current function)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation update (if none of the other choices apply)

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/RocketChat/Rocket.Chat/blob/develop/.github/CONTRIBUTING.md#contributing-to-rocketchat) doc
- [x] I have signed the [CLA](https://cla-assistant.io/RocketChat/Rocket.Chat.ReactNative)
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works (if applicable)
- [ ] I have added necessary documentation (if applicable)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: app/lib/rocketchat/services/shareExtension.ts [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+import log from '../../../utils/log';
+import { IShareServer, IShareUser } from '../../../reducers/share';
+import UserPreferences from '../../userPreferences';
+import database from '../../database';
+import RocketChat from '..';
+import { encryptionInit } from '../../../actions/encryption';
+import { store } from '../../auxStore';
+import sdk from './sdk';
+
+export async function shareExtensionInit(server: string) {
```
Comment (diegolmello) [inline app/lib/rocketchat/services/shareExtension.ts:14]: - Open Photos on iOS sim
- Try to share a photo to our app
- Share extension opens, but Reactotron raises this error
image

**Review by diegolmello — CHANGES_REQUESTED** [review]

File: app/definitions/ICredentials.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+export interface ICredentials {
+	password: string;
+	username: string;
+	email?: string;
+	ldap?: boolean;
+	ldapOptions?: object;
+}
```
Comment (gerzonc) [inline app/definitions/ICredentials.ts:7]: Comes from `@rocket.chat/sdk`

File: app/lib/rocketchat/services/sdk.ts [hunk 3]
```diff
@@ -158,6 +159,10 @@ class Sdk {
 	onStreamData(...args: any[]) {
 		return this.sdk.onStreamData(...args);
 	}
+
+	login(credentials: ICredentials) {
+		return this.sdk.login(credentials);
+	}
```
Comment (gerzonc) [inline app/lib/rocketchat/services/sdk.ts:165]: Needed; even though is something being worked on the dehydration for `login` methods, without this `shareExtensionInit` might crash like on previous review

File: app/definitions/IRocketChat.ts [hunk 4]
```diff
@@ -4,6 +4,7 @@ type TRocketChat = typeof rocketchat;
 
 export interface IRocketChat extends TRocketChat {
 	sdk: any;
+	shareSDK: any;
```
Comment (GleidsonDaniel) [inline app/definitions/IRocketChat.ts:7]:
```suggestion
```

File: app/lib/rocketchat/services/sdk.ts [hunk 5]
```diff
@@ -158,6 +159,10 @@ class Sdk {
 	onStreamData(...args: any[]) {
 		return this.sdk.onStreamData(...args);
 	}
+
+	login(credentials: ICredentials) {
```
Comment (GleidsonDaniel) [inline app/lib/rocketchat/services/sdk.ts:163]: Use current

File: app/lib/rocketchat/services/shareExtension.ts [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+	try {
+		const certificate = await UserPreferences.getStringAsync(`${RocketChat.CERTIFICATE_KEY}-${server}`);
+		if (SSLPinning && certificate) {
+			await SSLPinning.setCertificate(certificate, server);
+		}
+	} catch {
+		// Do nothing
+	}
+
+	this.shareSDK = sdk.disconnect();
```
Comment (GleidsonDaniel) [inline app/lib/rocketchat/services/shareExtension.ts:27]: I think there are other better ways to do this implementation without using this.

**Review by GleidsonDaniel — CHANGES_REQUESTED** [review]

**Review by GleidsonDaniel — APPROVED** [review]

**Pull request merged**